### ISSUE TITLE
Do i have to specify ethereal in my pickit code?

### ISSUE THREAD

**Issue by Jakfox91** [kolton/d2bot-with-kolbot#877]
So there are lines like the example here: 
[Name] == Demonhead && [Quality] == unique && [Flag] == ethereal # [EnhancedDefense] >= 100 && [LifeLeech] >= 8 && [Strength] >= 25 // Andariel's Visage

And then there are other lines like: 

[Name] == Shako && [Quality] == unique # [Defense] >= 98 // Harlequin Crest

So my question lies in the: ''[Flag] == ethereal'' part.
I know that the code uses the code == for it to BE etheareal to pick it up. 
And that you will have to write != for it to NOT be etheareal to be picked up.

But in the example of shako it doesnt specify anything, will any shako no matter it being ethereal or not be picked up then?

**Comment by DarkHorseDre** [comment]
any shako no matter it being ethereal or not will be picked up

**Comment by wills219** [comment]
Not sure if this answers your question but != ethereal means don't pick up eth and == means pick up

**Comment by Tubs557** [comment]
I am also wondering about.  I always assumed what DarkHorseDre said but can any confirm it?

**Comment by DarkHorseDre** [comment]
yes,  `[flag] != ethereal` will not save ethereal items.

The pickit tests each line explicitly and matches upon what it finds and nothing else.

so if you have both lines:
`[name] == shako && [quality] == unique`
`[name] == shako && [quality] == unique && [flag] == ethereal`
pickit will store both.
But equally, you can have 1 line to accomplish both:
`[name] == shako && [quality] == unique`
 - if the name of item is 'shako' and quality is 'unique' keep (regardless of etherealness as this was not specified. I am a script that does what I am told and nothing else!  A moronic processor).

`Ethereal` is a value of parameter `flag`. 
unid is/was a parameter but I cannot access my files from where i am and do not see any references online for it atm.  ultimately, if you can store an item based upon base stats without id'ing, then bot wont id.  usually you specify item type or name and quality type and bot will keep them all.

So, if you want to shako that has a certain value against a parameter which is only known when id'd, then bot will id to ascertain the value. example:
`[Name] == Shako && [Quality] == unique # [Defense] >= 98`
to know that the defense is >= 98, it needs to be id'd, and will be every-time.

To pick unid shako, *simply specify the base item and nothing else. bot will store what it finds (item named shako and quality unique - both known without id'ing):
`[Name] == Shako && [Quality] == unique`

*I swear there was a parameter for unid but cant find right now and may be prior to koltot. it doesn't matter as what I said above works.

**Comment by BabyStewStew** [comment]
So forgive my ignorance but does the ! cause it to not pick up ethereal items or to pick up both eth and non eth? I am trying to pickit Vortex shields both eth and non eth.

**Comment by DarkHorseDre** [comment]
'!=' means 'not equal to

if you want to pickup both, don't mention eth and it will pick both

e.g.  '[Name] == Shako && [Quality] == unique'
both eth and non-eth will be selected as nothing in that line mentions that flag.

**Comment by BabyStewStew** [comment]
Ok, thanks! I wanted to avoid any bot confusion so listed it as such: [name] == vortexshield && [quality] <= superior # [sockets] == 4 # [maxquantity] == 2. That should do the trick I would assume?

**Comment by DarkHorseDre** [comment]
yes - will pick eth and non-eth vortex shields superior or less with 4 sox (will also pick vortex with or without any res or any other stat as you havent listed it)

**Comment by BabyStewStew** [comment]
Perfect, thanks for breaking it down Barney style for me!

**Comment by DarkHorseDre** [comment]
lol nah it's not that mate - just helping you get the fundamentals.  once you get that you can expand it and build your own pickits as it's not code, just notation/logic.  GL!
